### PR TITLE
Added solution to SPR-04

### PR DESCRIPTION
**Pull request by MichaelShevchenko**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -23,6 +24,21 @@
             <artifactId>log4j</artifactId>
             <version>1.2.17</version>
         </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-context</artifactId>
+            <version>5.2.2.RELEASE</version>
```
Comment (dariiabutorina) [inline pom.xml]: Version attribute

**Review by dariiabutorina — COMMENTED** [review]
Good job!

File: pom.xml (hunk 1 above)
Comment (MichaelShevchenko) [reply pom.xml]: done

File: src/main/java/com/dev/cinema/dao/GenericDaoImpl.java [hunk 2]
```diff
@@ -1,20 +1,29 @@
...
 import org.apache.log4j.Logger;
 import org.hibernate.Session;
+import org.hibernate.SessionFactory;
 import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
 
+@Repository
 public class GenericDaoImpl<T> implements GenericDao<T> {
     private static final Logger logger = Logger.getLogger(OrderDaoImpl.class);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/GenericDaoImpl.java]:
```suggestion
    private static final Logger logger = Logger.getLogger(GenericDaoImpl.class);
```

Comment (MichaelShevchenko) [reply src/main/java/com/dev/cinema/dao/GenericDaoImpl.java]: done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**